perception/probabilistic_occupancy_grid_map/include/pointcloud_based_occupancy_grid_map/occupancy_grid_map.hpp

```diff
@@ -70,10 +71,15 @@ class OccupancyGridMap : public nav2_costmap_2d::Costmap2D
   OccupancyGridMap(
     const unsigned int cells_size_x, const unsigned int cells_size_y, const float resolution);
 
-  void updateWithPointCloud(
+  void updateWithPointCloudSimple(
     const PointCloud2 & raw_pointcloud, const PointCloud2 & obstacle_pointcloud,
     const Pose & robot_pose, const Pose & scan_origin);
 
+  void updateWithPointCloud3D(
+    const PointCloud2 & raw_pointcloud, const PointCloud2 & obstacle_pointcloud,
+    const Pose & robot_pose, const Pose & scan_origin, const double projection_dz_threshold,
+    const double obstacle_separation_threshold, grid_map::GridMap * debug);
+
```

Comment:
Can you define interface classes and have them inherit? (like strategy pattern)

Reply:
@yukkysaito Originally I created a new class `projective_pointcloud_based_occupancy_grid_map` and specified that from the launcher. Do you mean this mechanism ?